backend/server.js

```diff
@@ -69,6 +69,23 @@ async function updateUserProfileToDb(uri_, user_info) {
   }
 }
 
+async function validateUserProfileToDb(uri_, user_info) {
```

Comment:
You are missing however the code logic for when the user inputs incorrect credentials.

Reply:
Actually, it is handled with the throw catch statements.